osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetActualPowerQualityCommandExecutorTest.java

```diff
@@ -0,0 +1,169 @@
+
+        doReturn(generateMockedResult(attributeNames, AccessResultCode.SUCCESS)).when(this.dlmsHelper).getAndCheck(eq(this.conn), eq(this.dlmsDevice),
+                eq("retrieve actual power quality"), any(AttributeAddress.class));
+
+        final GetActualPowerQualityCommandExecutor executor =
+                new GetActualPowerQualityCommandExecutor(this.dlmsHelper);
+
+
+        final ActualPowerQualityResponseDto responseDto = executor.execute(this.conn, this.dlmsDevice,
+                this.actualPowerQualityRequestDto);
+
+        assertThat(responseDto.getActualPowerQualityData().getActualValues().size()).isEqualTo(attributeNames.size());
+        assertThat(responseDto.getActualPowerQualityData().getPowerQualityObjects().size()).isEqualTo(attributeNames.size());
+
+        for (int i=0;i<attributeNames.size();i++) {
+            final GetActualPowerQualityCommandExecutor.ActualPowerQualityAttributeName logicalName = attributeNames.get(i);
+
+            Serializable expectedValue;
+            String expectedUnit = null;
+            switch(logicalName.getClassId()) {
+                case CLASS_ID_CLOCK:     expectedValue = DateTime.parse("2018-12-31T23:00:00Z").toDate();
+                                         break;
+                case CLASS_ID_REGISTER:  expectedValue = BigDecimal.valueOf(i * 10.0);
+                                         expectedUnit = DlmsUnitTypeDto.VOLT.getUnit();
+                                         break;
+                case CLASS_ID_DATA:      expectedValue = BigDecimal.valueOf(i);
+                                         break;
+                default:                 expectedValue = null;
+                
+            }
+
+            final PowerQualityObjectDto powerQualityObjectDto = responseDto.getActualPowerQualityData().getPowerQualityObjects().get(i);
+            assertThat(powerQualityObjectDto.getName()).isEqualTo(logicalName.name());
+            assertThat(powerQualityObjectDto.getUnit()).isEqualTo(expectedUnit);
+
+            final ActualValueDto actualValue = responseDto.getActualPowerQualityData().getActualValues().get(i);
+            assertThat(actualValue.getValue()).isEqualTo(expectedValue);
+        }
+    }
+
+    private List<GetResult> generateMockedResult(
+            List<GetActualPowerQualityCommandExecutor.ActualPowerQualityAttributeName> attributeNames, AccessResultCode resultCode) {
+        return generateMockedResult(attributeNames, resultCode, DataObject.newDateTimeData(new CosemDateTime(2018, 12,
+                31, 23, 0, 0,
+                0)));
+    }
+
+    private List<GetResult> generateMockedResult(
+            List<GetActualPowerQualityCommandExecutor.ActualPowerQualityAttributeName> attributeNames,
+            AccessResultCode resultCode, DataObject dateTimeDataObject) {
+        List<GetResult> results = new ArrayList<>();
+
+        int idx = 1;
+        for (ActualPowerQualityAttributeName logicalName : attributeNames) {
```

Comment:
Missing final modifier with `logicalName` required by coding conventions.

As stated elsewhere the name `logicalName` or `attributeName` does not look appropriate for the gathered data with power quality related objects.

Reply:
Added final modifier loop variable logicalName renamed to metadata